### PR TITLE
Add show version parser for junos

### PR DESCRIPTION
**Pull request by sfloresk**
Added class ShowVersionSchema and class ShowVersion for cli command "show version"

### CODE REVIEW

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 1]
```diff
@@ -94,5 +94,44 @@ def test_golden_3(self):
...
+# ==========================
+class test_show_version(unittest.TestCase):
+    device = Device(name='aDevice')
+
+    golden_parsed_output = {'hostname': "JunosHostname-1",
+                             'operating_system': "Junos",
+                             'software_version': "18.2R2-S1",
+                             'model': "ex4200-24p",
+                             }
+
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:109]:
```
{
        "software-information": {
            "host-name": "JunosHostname-1", 
            "junos-version": "18.2R2-S1",
            "product-model": "ex4200-24p", 
            "product-name": "ex4200-24p"
            "package-information": 
                   [
                {
                    "comment": "JUNOS FIPS mode utilities [18.2R2-S1]", 
                    "name": "fips-mode-utilities"
                }, 
                {
                    "comment": "JUNOS EX  Software Suite [18.2R2-S1]", 
                    "name": "ex-software-suite"
                }, 
                (omit)
                   ]
}

```

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 2]
```diff
@@ -94,5 +94,44 @@ def test_golden_3(self):
...
+        JUNOS EX  Software Suite [18.2R2-S1]
+        JUNOS FIPS mode utilities [18.2R2-S1]
+        JUNOS Crypto Software Suite [18.2R2-S1]
+        JUNOS Online Documentation [18.2R2-S1]
+        JUNOS Phone-Home Software Suite [18.2R2-S1]
+        
+        {master:0}
+        user1@JunosHostname-1> 
+        '''}
+
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:126]: please add `def test_empty`,  here's an [example](https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/asa/tests/test_show_arp.py#L75)

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 3]
```diff
@@ -94,5 +94,44 @@ def test_golden_3(self):
         parsed_output = obj.parse(filename='filename999')
         self.assertEqual(parsed_output, self.golden_parsed_output3)
 
+
+# ==========================
+# Unit test for 'show version'
+# ==========================
+class test_show_version(unittest.TestCase):
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:101]: based on pep8, class name would be `TestShowVersion`

File: src/genie/libs/parser/junos/show_platform.py [hunk 4]
```diff
@@ -113,3 +114,70 @@ def cli(self, filename='', output=None):
...
+
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            m = p1.match(line)
+            if m:
+                show_version_dict['operating_system'] = 'Junos'
+                show_version_dict['software_version'] = \
+                    str(m.groupdict()['software_version'])
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/show_platform.py:167]: no need to use `str()` to cast the value, which defaults a string

File: src/genie/libs/parser/junos/show_platform.py [hunk 5]
```diff
@@ -113,3 +114,70 @@ def cli(self, filename='', output=None):
...
+        else:
+            out = output
+
+        # Init vars
+        show_version_dict = {}
+
+        # regex patterns
+
+        # Junos: 15.1R1-S1
+        p1 = re.compile('\s*Junos: +(?P<software_version>.*$)')
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/show_platform.py:151]: after `line.strip()`, there is no space at the beginning and the end of the line
so the regex usually looks like `re.compile(r'^Junos: +(?P<junos_version>\S+)$')`

please update other regexes as well

File: changelog/undistributed.rst [hunk 6]
```diff
@@ -171,8 +174,3 @@
         * show interfaces {interface} terse
     * Update ShowInterfacesTerseInterface
         * show interfaces terse {interface}
-
---------------------------------------------------------------------------------
-                                common.py
---------------------------------------------------------------------------------
-* updated _find_command to find command for nxos in aci mode
```
Comment (xiaoxinz-cisco) [inline changelog/undistributed.rst:178]: what's happening to this deleted line

File: src/genie/libs/parser/junos/show_platform.py [hunk 7]
```diff
@@ -113,3 +114,70 @@ def cli(self, filename='', output=None):
 
         return ret_dict
 
+# ===========================
+# Parser for show version
+# ===========================
+
+class ShowVersionSchema(MetaParser):
+    """Schema for show version"""
+    schema = {'hostname': str,
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/show_platform.py:123]: I added the desired parsed output below, please update the schema based on that

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 8]
```diff
@@ -10,30 +10,30 @@
 
 # Parser
 from genie.libs.parser.junos.show_platform import FileList
+from genie.libs.parser.junos.show_platform import ShowVersion
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:13]: `from genie.libs.parser.junos.show_platform import FileList, ShowVersion`

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 9]
```diff
@@ -94,5 +94,44 @@ def test_golden_3(self):
...
+        JUNOS Crypto Software Suite [18.2R2-S1]
+        JUNOS Online Documentation [18.2R2-S1]
+        JUNOS Phone-Home Software Suite [18.2R2-S1]
+        
+        {master:0}
+        user1@JunosHostname-1> 
+        '''}
+
+    def test_golden(self):
+        self.maxDiff = None
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:128]: please move it to the class level

File: src/genie/libs/parser/junos/tests/test_show_platform.py [hunk 10]
```diff
@@ -94,5 +94,44 @@ def test_golden_3(self):
         parsed_output = obj.parse(filename='filename999')
         self.assertEqual(parsed_output, self.golden_parsed_output3)
 
+
+# ==========================
+# Unit test for 'show version'
+# ==========================
+class test_show_version(unittest.TestCase):
+    device = Device(name='aDevice')
+
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/tests/test_show_platform.py:103]: please add `maxDiff = None` in here

File: changelog/undistributed.rst (hunk 6 above)
Comment (sfloresk) [reply changelog/undistributed.rst:178]: Not sure, I will re-add it

File: src/genie/libs/parser/junos/show_platform.py [hunk 11]
```diff
@@ -103,13 +105,125 @@ def cli(self, filename='', output=None):
...
+class ShowVersionSchema(MetaParser):
+    """Schema for show version"""
+    schema = \
+        {"software-information":
+            {
+                'host-name': str,
+                'junos-version': str,
+                'product-model': str,
+                'product-name': str,
+                'package-information': list
```
Comment (xiaoxinz-cisco) [inline src/genie/libs/parser/junos/show_platform.py:132]: informed on webex

**Pull request closed without merging**